core-client/transports/src/lib.rs

```diff
@@ -225,29 +183,34 @@ impl From<RpcChannel> for RawClient {
 
 impl RawClient {
 	/// Call RPC method with raw JSON.
-	pub fn call_method(&self, method: &str, params: Params) -> impl Future<Item = Value, Error = RpcError> {
+	pub fn call_method(&self, method: &str, params: Params) -> impl Future<Output = RpcResult<Value>> {
 		let (sender, receiver) = oneshot::channel();
 		let msg = CallMessage {
 			method: method.into(),
 			params,
 			sender,
 		};
-		self.0
-			.send(msg.into())
-			.map_err(|error| RpcError::Other(error.into()))
-			.and_then(|_| RpcFuture::new(receiver))
+		match self.0.send(msg.into()) {
+			Ok(()) => future::Either::Left(async move {
+				match receiver.await {
+					Ok(v) => v,
+					Err(e) => Err(RpcError::Other(e.into())),
+				}
+			}),
+			Err(error) => future::Either::Right(future::ready(Err(RpcError::Other(error.into())))),
+		}
```

Comment:
I did try making the whole method `async`, but the issue is with the lifetimes. The required future must be `'static`, and afaict there is no syntax to express that if you have `async fn`. So instead I rewrote the part that is really "async" into an `async move` block, but the outer function still has `'static` bound to prevent issues.